### PR TITLE
Feat/added msal integration in back

### PR DESCRIPTION
**Pull request by SandrineCipolla**

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -21,13 +21,17 @@
...
+    "passport": "^0.7.0",
+    "passport-azure-ad": "^4.3.5",
     "react-router-dom": "^6.21.1"
   },
   "devDependencies": {
     "@types/cors": "^2.8.17",
     "@types/express": "^4.17.21",
     "@types/jest": "^29.5.12",
     "@types/mysql": "^2.15.25",
+    "@types/passport": "^1.0.16",
```
Comment (macreiben-dev) [inline backend/package.json]: issue(blocking): depencies duplicate with prod dependencies for passport related libs.

File: backend/src/authConfig.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const passportConfig = {
+    credentials: {
+        tenantName: 'stockhubb2c.onmicrosoft.com',
+        clientID: 'dc30ef57-cdc1-4a3e-aac5-9647506a72ef',
```
Comment (macreiben-dev) [inline backend/src/authConfig.ts]: issue(blocking): should not expose credential related data in git repo.

File: frontend/src/authConfig.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        signUpSignIn: {
+           authority: 'https://stockhubb2c.b2clogin.com/stockhubb2c.onmicrosoft.com/B2C_1_signupsignin',
+            //authority: "https://stockhubb2c.b2clogin.com/tfp/2b9bbbf5-4706-4eed-86a7-d55a5094b647/B2C_1_signupsignin",
+        },
+        // forgotPassword: {
+        //     authority: 'https://fabrikamb2c.b2clogin.com/fabrikamb2c.onmicrosoft.com/B2C_1_reset_v3',
+        // },
+        // editProfile: {
+        //     authority: 'https://fabrikamb2c.b2clogin.com/fabrikamb2c.onmicrosoft.com/b2c_1_edit_profile_v2',
+        // },
```
Comment (macreiben-dev) [inline frontend/src/authConfig.ts]: issue(blocking): commented code to remove

File: frontend/src/components/StocksList.tsx [hunk 4]
```diff
@@ -19,8 +25,9 @@ const StocksList: React.FC = () => {
     }, []);
 
     return (
+<AuthenticatedTemplate>
         <div>
-            <h2 className="text-lg font-bold mb-2 mt-2">Liste des stocks</h2>
+            <h2 className="text-lg font-bold mb-2 mt-2">Liste des stocks {msalInstance.accounts[0].username}</h2>
```
Comment (macreiben-dev) [inline frontend/src/components/StocksList.tsx]: suggestion: use a wrapper around the msal instance to hide the .account[0].username.

**Review by macreiben-dev — APPROVED** [review]

File: frontend/src/authConfig.ts (hunk 3 above)
Comment (SandrineCipolla) [reply frontend/src/authConfig.ts]: je garde car après on va mettre le profil

File: frontend/src/components/StocksList.tsx (hunk 4 above)
Comment (SandrineCipolla) [reply frontend/src/components/StocksList.tsx]: à voir sur la branche front

**Pull request merged**